### ISSUE TITLE
Fix macOS test failures

### ISSUE THREAD

**Issue by riccardoporreca** [miraisolutions/rTRNG#15]
Revealed by the new GH Actions workflow for `macOS-latest (oldrel)` (3.6.3) => https://github.com/miraisolutions/rTRNG/runs/1765618265?check_suite_focus=true 

```
[ FAIL 30 | WARN 0 | SKIP 0 | PASS 581 ]
```

All errors are of the form
```
 ── Failure (test-TRNG.Random.R:135:3): TRNGsplit errors for out-of-range subsequence indices ──
`TRNGsplit(p, p + 1L)` threw an error with unexpected class.
Expected class: std::invalid_argument
Actual class:   simpleError/error/condition
Message:        c++ exception (unknown reason)
```

and correspond to the current `r-oldrel-macos-x86_64` for the released package version 4.20-1 on CRAN (in [CRAN checks](https://cran.r-project.org/web/checks/check_results_rTRNG.html)).

This reveals the approach in #10 (e4ac407) is not working on this platform and should be addressed differently.

**Comment by riccardoporreca** [comment]
This is most likely related to what described in RcppCore/Rcpp#972.

We need to work around the fact that, in certain circumstances (architectures / compilers), we cannot rely on errors propagating with proper `std::invalid_argument` class / error message.

**Comment by riccardoporreca** [comment]
Fixed in b292e3a
